src/main/java/spireMapOverhaul/zones/thefog/relics/StoneFragment.java

```diff
@@ -13,30 +15,33 @@ public class StoneFragment extends AbstractSMORelic {
 
     public StoneFragment() {
         super(ID, TheFogZone.ID, RelicTier.SPECIAL, LandingSound.HEAVY);
+        counter = 13;
     }
 
     @Override
     public void atTurnStart() {
-        pulse = true;
-        beginPulse();
-        counter = 0;
+        if (counter < 1) return;
+        flash();
+        atb(new RelicAboveCreatureAction(AbstractDungeon.player, this));
+        atb(new GainEnergyAction(1));
+        counter--;
+        if (counter < 1) {
+            counter = -1;
+            usedUp();
+        }
     }
 
     @Override
-    public void onCardDraw(AbstractCard drawnCard) {
-        counter += 1;
-        if (counter == 6) {
-            flash();
-            pulse = false;
-            stopPulse();
-            atb(new GainEnergyAction(1));
-        }
+    public void update() {
+        super.update();
+        description = (counter < 1) ? MSG[2] : DESCRIPTIONS[1]; // MSG[2] is the "used up" string
```

Comment:
I guess you do this so it works correctly on save and load? Maybe we can find an alternative like checking justEnteredRoom + onVictory or something. I generally like to avoid doing something every frame if it's not needed.

Reply:
Just committed some replacement code - thoughts?